go/private/actions/stdlib.bzl

```diff
@@ -63,24 +63,46 @@ def _should_use_sdk_stdlib(go):
             not go.mode.gc_goopts and
             go.mode.link == LINKMODE_NORMAL)
 
-def _build_stdlib_list_json(go):
+def _build_stdlib_list_json(go, env):
     out = go.declare_file(go, "stdlib.pkg.json")
     args = go.builder_args(go, "stdliblist")
     args.add("-sdk", go.sdk.root_file.dirname)
     args.add("-out", out)
+
     go.actions.run(
         inputs = go.sdk_files,
         outputs = [out],
         mnemonic = "GoStdlibList",
         executable = go.toolchain._builder,
         arguments = [args],
-        env = go.env,
+        env = env,
     )
     return out
 
+def _cgo_env(go):
+    env = go.env
+
+    # NOTE(#2545): avoid unnecessary dynamic link
+    # go std library doesn't use C++, so should not have -lstdc++
+    # Also drop coverage flags as nothing in the stdlib is compiled with
+    # coverage - we disable it for all CGo code anyway.
+    ldflags = [
+        option
+        for option in extldflags_from_cc_toolchain(go)
+        if option not in ("-lstdc++", "-lc++") and option not in COVERAGE_OPTIONS_DENYLIST
+    ]
+    env.update({
+        "CGO_ENABLED": "1",
+        "CC": go.cgo_tools.c_compiler_path,
+        "CGO_CFLAGS": " ".join(go.cgo_tools.c_compile_options),
+        "CGO_LDFLAGS": " ".join(ldflags),
+    })
+
+    return env
+
 def _sdk_stdlib(go):
     return GoStdLib(
-        _list_json = _build_stdlib_list_json(go),
+        _list_json = _build_stdlib_list_json(go, _cgo_env(go)),
```

Comment:
This means we would enable cgo unconditionally here, even for targets built with cgo disabled. Can you gate on `go.mode.pure`?

Reply:
Isn't the SDK always built with CGO?

Reply:
Also, `_build_stdlib()` sets `CGO_ENABLED=0` when `go.mode.pure`. To be consistent, this function should also set that.

Reply:
Ahhha! That explains a lot for me. Thank you =) I'll get the `go.mod.pure` gate in place!